data-node/src/main/java/org/graylog/datanode/configuration/variants/MongoCertSecureConfiguration.java

```diff
@@ -86,7 +86,7 @@ public MongoCertSecureConfiguration(final Configuration localConfiguration,
         this.secret = passwordSecret.toCharArray();
 
         //TODO: matches line 123 of DataNodePreflightGeneratePeriodical, but both need to be changed
-        this.mongoKeystorePassword = localConfiguration.getDatanodeHttpCertificatePassword();
+        this.mongoKeystorePassword = localConfiguration.getDatanodeHttpCertificatePassword() != null ? localConfiguration.getDatanodeHttpCertificatePassword().toCharArray() : secret;
```

Comment:
The same as in previous comment - should not this password be always equal to secret?

Reply:
I believe I have fixed that.